core/src/main/scala/cromwell/core/WorkflowOptions.scala

```diff
@@ -179,18 +179,36 @@ case class WorkflowOptions(jsObject: JsObject) {
 
   /**
     * Returns a JSON representation of these workflow options where the encrypted values
-    * have been replaced by the string "cleared". This will be called on the workflow
-    * options (and subsequently stored back in the database) once a workflow finishes
-    * and the encrypted values aren't needed anymore. This protects us in case the
-    * database and private key become compromised, the attacker will not be able to
-    * decrypt values for completed workflows.
+    * have been replaced by the string "cleared".
+    *
+    * Used to protect encrypted values from being stored in metadata
     */
-  def clearEncryptedValues: String = {
-    val revoked = jsObject.fields map {
-      case (k, v: JsObject) if isEncryptedField(v) => k -> JsString("cleared")
-      case (k, v) => k -> v
+  def clearEncryptedValues: WorkflowOptions = {
+
+    def revoke(o: JsObject): JsObject = {
+      val newFields = o.fields map {
+        case (k, v: JsObject) if isEncryptedField(v) => k -> JsString("cleared")
+        case (k, v: JsObject) => k -> revoke(v)
+        case (k, v) => k -> v
+      }
+      JsObject(newFields)
+    }
+
+    WorkflowOptions(revoke(jsObject))
+  }
+
+  def inferHogGroup(workflowId: WorkflowId): String = {
+    val config = ConfigFactory.load
```

Comment:
> hog groups seem to be an backend construct before this PR (though I'm not sure why it's hosted there)

* It was definitely a misleading location. It was actually (shared) code in the backend descriptor constructor that decided the hog factor and until now, only the job execution actor in the engine which used the hog factor. Going back to the original implementation, I think I wanted this to be more widely useable than just "job hogging" so wanted this value to be available in the backend too.

> the groups had no classes / structures / first-class support in core, just a copy-pastable block of code

* That's a good point. I can't decide whether logically the hog factor is "an inferable part of the workflow options" or just "a function of workflow options and workflow ID". And if the latter, should it be a function or memoized somewhere? Either way, finding a home for Hog Factors and making only a one-off call to config at the start of time seems like a good idea.

> post PR the core implementation of hog group is read by the engine, instead of using the backend version

Correct - that's be design (see my comment below):

> none of the implementations afaict pass in the hog-group-field -- each time it's re-looked up from the raw `ConfigFactory.load`

This should be fixed. The backend descriptor shouldn't be re-calculating this, there's no reason not to reuse the workflow options